DigitalLearningSolutions.Web/ViewModels/MyAccount/MyAccountViewModel.cs

```diff
@@ -1,19 +1,39 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.MyAccount
 {
+    using DigitalLearningSolutions.Data.Models.CustomPrompts;
     using DigitalLearningSolutions.Data.Models.User;
 
     public class MyAccountViewModel
     {
         public MyAccountViewModel(
             AdminUser? adminUser,
-            DelegateUser? delegateUser)
+            DelegateUser? delegateUser,
+            CentreCustomPrompts? customPrompts)
         {
             FirstName = adminUser?.FirstName ?? delegateUser?.FirstName;
             Surname = adminUser?.LastName ?? delegateUser?.LastName;
             User = adminUser?.EmailAddress ?? delegateUser?.EmailAddress;
             ProfilePicture = adminUser?.ProfileImage ?? delegateUser?.ProfileImage;
             Centre = adminUser?.CentreName ?? delegateUser?.CentreName;
             DelegateNumber = delegateUser?.CandidateNumber;
+            JobGroup = delegateUser?.JobGroupName;
+
+            CustomField1 = PopulateCustomFieldViewModel(customPrompts?.CustomField1, delegateUser?.Answer1);
+            CustomField2 = PopulateCustomFieldViewModel(customPrompts?.CustomField2, delegateUser?.Answer2);
+            CustomField3 = PopulateCustomFieldViewModel(customPrompts?.CustomField3, delegateUser?.Answer3);
+            CustomField4 = PopulateCustomFieldViewModel(customPrompts?.CustomField4, delegateUser?.Answer4);
+            CustomField5 = PopulateCustomFieldViewModel(customPrompts?.CustomField5, delegateUser?.Answer5);
+            CustomField6 = PopulateCustomFieldViewModel(customPrompts?.CustomField6, delegateUser?.Answer6);

```

Comment:
From what I can tell, this is the first point where the answers and prompts are linked together in a single object. Would it be worth handling these multiple CustomFields from this point onwards in a list or an array or something?
I think before this the field numbers have to be preserved as they come from different tables and the ordering matters, but a from here we might be able to cut down on nearly-identical code with looping. Or  would we prefer to keep it to a specific number order to keep it inline with the database?

Reply:
Ah, just saw this comment - I think my mind went the same direction below. :)

Reply:
I've had a look in the database, and we have a whole load of centres where the custom fields IDs are not populated in order, e.g. the IDs are 0,1,5 for prompts 1,2,3 respectively. This would give us null for Prompt1 currently, I'm not sure whether it would be better to keep it as the separate fields or add some kind of property to the view model to signify which field it is. I'm not sure which will be better when we are going in the other direction for Edit Details. Thoughts?

Reply:
I think adding a property to the view model (or even a CustomPrompt that could be used here directly) to signify which field it is would be useful - it would work well with both display & edit pages and will keep the views simple too.

A bit separate point, but now that I think about it, do we even need the customPromptId as when it's 0, the custom field name should be null on join?

Reply:
OK will refactor it that way instead, might end up needing to modify it a bit later for the Edit screen, but will deal with that when I get there

Reply:
And yes, we shouldn't need the CustomPromptIds